mitodl/micromasters
# Support local SSO with edx devstack

#### Pre-Flight checklist


#### What's this PR do?
This allows developers to more easily get OAuth SSO working between Micromasters and edX (Devstack) when running both locally.  As part of this change, I am using `EDXORG_BASE_URL` as the outbound base URL for edX and `EDXORG_CALLBACK_URL` for the inbound base URL for edX which is why you will see a lot of the original instances of `EDXORG_BASE_URL` have now been changed to `EDXORG_CALLBACK_URL`.

#### How should this be manually tested?
Follow the updated instructions for setting up Micromasters.  If you already have Micromasters running locally, then replace the following with the values shown in your `.env` file:
```
EDXORG_CALLBACK_URL=http://host.docker.internal:18000/
EDXORG_BASE_URL=http://edx.odl.local:18000/
OPENEDX_HOST_ENTRY=localhost:127.0.0.1
```

Verify the functionality of this PR be ensuring that you can create a new Micromasters account via SSO with edX as well as login to an existing Micromasters account via SSO with edX.
